### PR TITLE
use workers in prerun mode

### PR DESCRIPTION
**Pull request by dboreham**
Adds worker mode to the pre-run code path, enabled with `prerun.parallel = true`. 
Also adds pprof profiling agent (used to capture goroutine stack snapshots from full index node diffing runs, for performance analysis).

### CODE REVIEW

File: pkg/service.go [hunk 1]
```diff
@@ -118,18 +118,55 @@ func (sds *Service) APIs() []rpc.API {
...
+			logrus.Infof("sequential processing prerun range (%d, %d)", preRun.Start, preRun.Stop)
+			for i := preRun.Start; i <= preRun.Stop; i++ {
+				if err := sds.WriteStateDiffAt(i, preRun.Params); err != nil {
+					return fmt.Errorf("error writing statediff at height %d in range (%d, %d) : %v", i, preRun.Start, preRun.Stop, err)
+				}
 			}
 		}
 	}
 	sds.preruns = nil
+	// At present this code is never called so we have not written the parallel version:
```
Comment (i-norden) [inline pkg/service.go:167]: Honestly can't recall when I was thinking the `rngs` argument might be used...

File: README.md [hunk 2]
```diff
@@ -122,6 +123,9 @@ An example config file:
     # path to custom chain config file (optional)
     # keep chainID same as that in chain config file
     chainConfig  = "./chain.json"           # ETH_CHAIN_CONFIG
+
+[debug]
+    pprof = false                           # Enable pprof agent listener on port 6060
```
Comment (i-norden) [inline README.md]: nitpick: add `# DEBUG_PPROF`

**Review by i-norden — APPROVED** [review]
LGTM!

File: README.md (hunk 2 above)
Comment (dboreham) [reply README.md]: Fixed.

**Pull request merged**